File: Cli_Core_JavaFX/src/cli_core_javafx_test/Cmd_Token.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cli_core_javafx_test;

/**
 *
 * @author mike
 */
public class Cmd_Token {

    protected String Text;

    public Cmd_Token(String text) {
        Text = text;
    }

    public String Text_Get() {
        return Text;
    }

    public String Text_Get_Without_COMMAS() {
        if (Text.length() >= 2 && Text.charAt(0) == '\"' && Text.charAt(Text.length() - 1) == '\"') {
            return Text.substring(1, Text.length() - 2);
        }
        if (Text.length() >= 2 && Text.charAt(0) == '\'' && Text.charAt(Text.length() - 1) == '\'') {
            return Text.substring(1, Text.length() - 2);
        }
        return Text;
    }

    public String To_String() {
        return "Text:" + Text;
    }

}

File: Cli_Core_CS/Cli_Core_CS/Cmd_Token.cs
namespace Cli_Core_CS
{
    class Cmd_Token
    {
        protected string Text;

        public Cmd_Token(string text)
        {
            Text = text;
        }

        public string Text_Get()
        {
            return Text;
        }

        public string Text_Get_Without_COMMAS()
        {
            if (Text.Length >= 2 && Text[0] == '\"' && Text[Text.Length - 1] == '\"')
            {
                return Text.Substring(1, Text.Length - 2);
            }
            if (Text.Length >= 2 && Text[0] == '\'' && Text[Text.Length - 1] == '\'')
            {
                return Text.Substring(1, Text.Length - 2);
            }
            return Text;
        }

        public string To_String()
        {
            return "Text:" + Text;
        }

    }
}
